src/Microsoft.AspNet.Mvc.Core/Internal/ResponseContentTypeHelper.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using Microsoft.AspNet.Mvc.Core;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc.Internal
+{
+    public static class ResponseContentTypeHelper
+    {
+        public static void ResolveContentTypeAndEncoding(
+            MediaTypeHeaderValue actionResultContentType,
+            string httpResponseContentType,
+            MediaTypeHeaderValue defaultContentType,
+            out string resolvedResponseContentType,
+            out Encoding resolvedContentTypeEncoding)
+        {
+            if (defaultContentType == null)
+            {
+                throw new ArgumentNullException(nameof(defaultContentType));
+            }
+
+            if (defaultContentType.Encoding == null)
+            {
+                throw new ArgumentException(
+                    Resources.FormatDefaultContentTypeMustHaveEncoding(defaultContentType.ToString()));
+            }
+
+            // 1. User sets the ContentType property on the action result
```

Comment:
These comments need to be beefed up. The original comments benefitted from the context of the surrounding code.